### DISCUSSION TITLE
Grasp/Object Pose Estimation

### DISCUSSION THREAD

**Discussion by peterdavidfagan** [moveit/moveit2#2102] [Q&A] [answered]
Hi all, 

I am working on object pose estimation as part of a demo I am creating that leverages MoveIt 2. I wished to ask what methods (and packages) others are using for pose estimation. I am especially interested in packages that are compatible with the ROS 2 ecosystem. I am also open to working on porting ROS packages to ROS 2. 

Thus far I am starting to explore object pose detection with: 

- https://github.com/NVlabs/Deep_Object_Pose

I am also looking at grasp pose detection: 

- https://github.com/NVlabs/contact_graspnet
- https://github.com/atenpas/gpd

Any feedback or advice would be much appreciated. I will look to open source the demo and post a link to it in this thread once it is completed.

**Comment by JafarAbdi** [comment, marked as answer]
You might find https://github.com/ros-planning/moveit_task_constructor/pull/196 and https://github.com/bostoncleek/deep_grasp_demo/tree/update_generators interesting, they're ROS 1 but aren't using it much, so porting them shouldn't take a long time

**Reply by peterdavidfagan** [reply]
Thanks @JafarAbdi.
